[Bug] rocketmq5.3.x, 开启 acl 2.0，消费者接收不到消息

### Before Creating the Bug Report

- [x] I found a bug, not just asking a question, which should be created in [GitHub Discussions](https://github.com/apache/rocketmq/discussions).

- [x] I have searched the [GitHub Issues](https://github.com/apache/rocketmq/issues) and [GitHub Discussions](https://github.com/apache/rocketmq/discussions)  of this repository and believe that this is not a duplicate.

- [x] I have confirmed that this bug belongs to the current repository, not other repositories of RocketMQ.


### Runtime platform environment

OS: Centos7
容器化部署

### RocketMQ version

5.3.2、5.3.3

### JDK Version

openjdk version "1.8.0_452"

### Describe the Bug

开启 acl2.0 之后，发送者正常发送，消费者无法消费消息

### Steps to Reproduce

1、Docker 部署 rocketmq，集群模式
broker.conf 配置文件：
![Image](https://github.com/user-attachments/assets/afac41d6-f38c-4bed-b485-0ee10a52b675)

rmq-proxy.json 配置文件
![Image](https://github.com/user-attachments/assets/cdc88786-f5d6-41ef-ae7e-e63aefee9327)

2、生产者发送消息
3、消费者订阅消息

### What Did You Expect to See?

开启 acl 2.0，消费者正常接收消息

### What Did You See Instead?

消费者接收不到消息

### Additional Context

_No response_

> [rocketmqOnly.zip](https://github.com/user-attachments/files/21401831/rocketmqOnly.zip)
> 
> emmmm我这里也是，有点抽象了，后面发现是没有给挂在的容器设置访问权限，然后我就出现了能发送消息但是没有可以消费的情况，不知道你是不是这种情况 容器内部是只允许rocketmq的用户有访问权限，但是我用容器映射进行绑定挂在的话，容器的权限就变成root了，然后rockemq用户就没有权限写入commitlog的文件取存储消息数据，这时候启动容器后还要给store目录下的允许rockemq用户进行操作 `chmod 777 /home/rocketmq/ -R && chown -R rocketmq:rocketmq /home/rocketmq/logs/ && chown -R rocketmq:rocketmq /home/rocketmq/store/ && ls -ld /home/rocketmq/store/ `
> 
> 也可能是rmq-proxy需要加这个"enableAclRpcHookForClusterMode": true,选项
> 
> 或许这个大佬的文章详细的可以给你一点参考https://blog.zcw159357.com/article/1/2025-03-15-0048.html
> 
> 我的配置文件参考，或许可以帮助到你

你好，请问一下使用docker部署rocketmq5.3.3版本服务集群，你的宿主机是什么操作系统？容器中使用的是rocketmq用户，如果挂载broker的store目录，执行容器内部授权操作之后，启动服务的时候挂载的store中新增的文件权限还是无法生效？报错：java.lang.RuntimeException: Failed to load the authentication metadata provider
	at org.apache.rocketmq.auth.authentication.factory.AuthenticationFactory.lambda$getMetadataProvider$1(AuthenticationFactory.java:89)
	at org.apache.rocketmq.auth.authentication.factory.AuthenticationFactory.computeIfAbsent(AuthenticationFactory.java:144)
	at org.apache.rocketmq.auth.authentication.factory.AuthenticationFactory.getMetadataProvider(AuthenticationFactory.java:78)
	at org.apache.rocketmq.auth.authentication.factory.AuthenticationFactory.getMetadataProvider(AuthenticationFactory.java:66)
	at org.apache.rocketmq.auth.authentication.manager.AuthenticationMetadataManagerImpl.<init>(AuthenticationMetadataManagerImpl.java:42)
	at org.apache.rocketmq.auth.authentication.factory.AuthenticationFactory.getMetadataManager(AuthenticationFactory.java:70)
	at org.apache.rocketmq.broker.BrokerController.<init>(BrokerController.java:362)
	at org.apache.rocketmq.broker.BrokerStartup.buildBrokerController(BrokerStartup.java:215)
	at org.apache.rocketmq.broker.BrokerStartup.createBrokerController(BrokerStartup.java:247)
	at org.apache.rocketmq.broker.BrokerStartup.main(BrokerStartup.java:52)
Caused by: java.lang.RuntimeException: Failed to load rocksdb for auth_user, please check whether it is occupied
	at org.apache.rocketmq.auth.authentication.provider.LocalAuthenticationMetadataProvider.initialize(LocalAuthenticationMetadataProvider.java:50)
	at org.apache.rocketmq.auth.authentication.factory.AuthenticationFactory.lambda$getMetadataProvider$1(AuthenticationFactory.java:86)
	... 9 more

--- a/broker/src/main/java/org/apache/rocketmq/broker/processor/NotificationProcessor.java
+++ b/broker/src/main/java/org/apache/rocketmq/broker/processor/NotificationProcessor.java
@@ -19,7 +19,6 @@ package org.apache.rocketmq.broker.processor;
 import io.netty.channel.Channel;
 import io.netty.channel.ChannelHandlerContext;
 import java.util.Map;
-import java.util.Objects;
 import java.util.Random;
 import org.apache.rocketmq.broker.BrokerController;
 import org.apache.rocketmq.broker.longpolling.PollingHeader;

--- a/broker/src/main/java/org/apache/rocketmq/broker/processor/NotificationProcessor.java
+++ b/broker/src/main/java/org/apache/rocketmq/broker/processor/NotificationProcessor.java
@@ -79,16 +79,17 @@ public class NotificationProcessor implements NettyRequestProcessor {
     @Override
     public RemotingCommand processRequest(final ChannelHandlerContext ctx,
         RemotingCommand request) throws RemotingCommandException {
-        request.addExtFieldIfNotExist(BORN_TIME, String.valueOf(System.currentTimeMillis()));
-        if (Objects.equals(request.getExtFields().get(BORN_TIME), "0")) {
-            request.addExtField(BORN_TIME, String.valueOf(System.currentTimeMillis()));
-        }
         Channel channel = ctx.channel();
 
         RemotingCommand response = RemotingCommand.createResponseCommand(NotificationResponseHeader.class);
         final NotificationResponseHeader responseHeader = (NotificationResponseHeader) response.readCustomHeader();
         final NotificationRequestHeader requestHeader =
-            (NotificationRequestHeader) request.decodeCommandCustomHeader(NotificationRequestHeader.class);
+            request.decodeCommandCustomHeader(NotificationRequestHeader.class, true);
+        if (requestHeader.getBornTime() == 0) {
+            final long beginTimeMills = this.brokerController.getMessageStore().now();
+            request.addExtField(BORN_TIME, String.valueOf(beginTimeMills));
+            requestHeader.setBornTime(beginTimeMills);
+        }
 
         response.setOpaque(request.getOpaque());
 

--- a/broker/src/main/java/org/apache/rocketmq/broker/processor/PopMessageProcessor.java
+++ b/broker/src/main/java/org/apache/rocketmq/broker/processor/PopMessageProcessor.java
@@ -228,18 +228,16 @@ public class PopMessageProcessor implements NettyRequestProcessor {
 
         final long beginTimeMills = this.brokerController.getMessageStore().now();
 
-        // fill bron time to properties if not exist, why we need this?
-        request.addExtFieldIfNotExist(BORN_TIME, String.valueOf(System.currentTimeMillis()));
-        if (Objects.equals(request.getExtFields().get(BORN_TIME), "0")) {
-            request.addExtField(BORN_TIME, String.valueOf(System.currentTimeMillis()));
-        }
-
         Channel channel = ctx.channel();
         RemotingCommand response = RemotingCommand.createResponseCommand(PopMessageResponseHeader.class);
         response.setOpaque(request.getOpaque());
 
         final PopMessageRequestHeader requestHeader =
             request.decodeCommandCustomHeader(PopMessageRequestHeader.class, true);
+        if (requestHeader.getBornTime() == 0) {
+            request.addExtField(BORN_TIME, String.valueOf(beginTimeMills));
+            requestHeader.setBornTime(beginTimeMills);
+        }
         final PopMessageResponseHeader responseHeader = (PopMessageResponseHeader) response.readCustomHeader();
 
         // Pop mode only supports consumption in cluster load balancing mode

--- a/broker/src/main/java/org/apache/rocketmq/broker/processor/PopMessageProcessor.java
+++ b/broker/src/main/java/org/apache/rocketmq/broker/processor/PopMessageProcessor.java
@@ -29,7 +29,6 @@ import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
 import java.util.Map.Entry;
-import java.util.Objects;
 import java.util.Random;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ConcurrentHashMap;

--- a/proxy/src/main/java/org/apache/rocketmq/proxy/service/message/LocalMessageService.java
+++ b/proxy/src/main/java/org/apache/rocketmq/proxy/service/message/LocalMessageService.java
@@ -199,7 +199,6 @@ public class LocalMessageService implements MessageService {
     @Override
     public CompletableFuture<PopResult> popMessage(ProxyContext ctx, AddressableMessageQueue messageQueue,
         PopMessageRequestHeader requestHeader, long timeoutMillis) {
-        requestHeader.setBornTime(System.currentTimeMillis());
         RemotingCommand request = LocalRemotingCommand.createRequestCommand(RequestCode.POP_MESSAGE, requestHeader, ctx.getLanguage());
         CompletableFuture<RemotingCommand> future = new CompletableFuture<>();
         SimpleChannel channel = channelManager.createInvocationChannel(ctx);

